### PR TITLE
Drop redundant constraint in `selectT`

### PR DESCRIPTION
**Pull request by snowleopard**
As pointed out by @turion in #53, we don't need it!

### CODE REVIEW

File: src/Control/Selective.hs [hunk 1]
```diff
@@ -196,10 +196,10 @@ branch x l r = fmap (fmap Left) x <*? fmap (fmap Right) l <*? r
 selectA :: Applicative f => f (Either a b) -> f (a -> b) -> f b
 selectA x y = (\e f -> either f id e) <$> x <*> y
 
--- | If a functor is both 'Applicative' and 'Traversable', we can implement
--- 'select' in another interesting way: the effects associated with the second
--- argument can be skipped as long as the first argument contains only 'Right's.
-selectT :: (Applicative f, Traversable f) => f (Either a b) -> f (a -> b) -> f b
+-- | For traversable functors, we can implement 'select' in another interesting
```
Comment (turion) [inline src/Control/Selective.hs:199]:
```suggestion
-- | For 'Traversable' 'Functor's, we can implement 'select' in another interesting
```

**Review by turion — APPROVED** [review]

**Comment by turion** [comment]
> @turion I've invited you to join as a collaborator so you can review PRs. Hope you don't mind!

Thanks a lot! Although, for reviewing I don't need to be collaborator. If you want me give me commit rights, it makes sense to add me.

File: src/Control/Selective.hs (hunk 1 above)
Comment (snowleopard) [reply src/Control/Selective.hs:199]: I think we tend to say just "applicative functors" in many comments already, so sticking to "traversable functors" seems more consistent.

**Comment by snowleopard** [comment]
> for reviewing I don't need to be collaborator

I don't think that's true: I couldn't add you as a reviewer for this PR before you joined as a collaborator.

Anyway, thanks for the review!

**Pull request merged**